File: frontend/src/app/model/ThreadCategoryDTO.ts
import { ThreadDTO } from "./ThreadDTO";

export interface ThreadCategoryDTO {
    encryptedId: string,
    name: string,
    description: string,
    subCategories: ThreadCategoryDTO[],
    numberOfThreads: number,
    numberOfPosts: number,
    newestThread: ThreadDTO,
    thumbnailPath: string
}

File: backend/dto/src/main/java/pl/dawid0604/pcForum/dto/thread/ThreadCategoryDTO.java
package pl.dawid0604.pcForum.dto.thread;

import java.util.List;

public record ThreadCategoryDTO(
    String encryptedId,
    String name,
    String description,
    long numberOfThreads,
    long numberOfPosts,
    String thumbnailPath,
    List<ThreadCategoryDTO> subCategories,
    ThreadDTO newestThread) { }
